client-telemetry-reporter-plugin/kafka/README.md

```diff
@@ -0,0 +1,159 @@
+
+### Build the client-telemetry-reporter-plugin
+
+Build the `client-telemetry-reporter-plugin` JAR.
+
+```shell
+./gradlew clean :client-telemetry-reporter-plugin:kafka:build
+```
+
+The above command builds the `client-telemetry-reporter-plugin` JAR located at:
+`client-telemetry-reporter-plugin/kafka/build/libs/client-telemetry-reporter-plugin.jar`
+
+### Apache Kafka Broker
+
+Download the Apache Kafka 3.7.0 or higher release from [here](https://kafka.apache.org/downloads).
+
+```shell
+wget https://downloads.apache.org/kafka/3.8.0/kafka_2.13-3.8.0.tgz
+tar -xvf kafka_2.13-3.8.0.tgz
+cd kafka_2.13-3.8.0
+```
+
+#### Add the client-telemetry-reporter-plugin to the Kafka Broker
+
+Copy the `client-telemetry-reporter-plugin.jar` to the Kafka broker `libs` directory.
+
+```shell
+cp <tutorials_repo_path>/tutorials/client-telemetry-reporter-plugin/kafka/build/libs/client-telemetry-reporter-plugin.jar <kafka_path>/kafka_2.13-3.8.0/libs/
+```
+
+#### Configure the Kafka Broker
+
+Add the following configuration to the Kafka broker `kraft` properties file `server.properties`,
+located at `<kafka_path>/kafka_2.13-3.8.0/config/kraft/server.properties`.
+
+```properties
+# Add the client telemetry reporter plugin
+metric.reporters=io.confluent.developer.ClientOtlpMetricsReporter
+```
+
+#### Start the Kafka Broker
+
+```shell
+KAFKA_CLUSTER_ID="$(bin/kafka-storage.sh random-uuid)"
+./bin/kafka-storage.sh format -t $KAFKA_CLUSTER_ID -c config/kraft/server.properties
+OTEL_EXPORTER_OTLP_ENDPOINT=127.0.0.1:4317 ./bin/kafka-server-start.sh config/kraft/server.properties
+```
+
+#### Create metrics subscription
+
+Create a metrics subscription to receive client metrics from the Kafka broker.
+
+```shell
+./bin/kafka-client-metrics.sh --bootstrap-server localhost:9092 --alter --name <subscription_name> --metrics <metric_name_prefix> --interval <interval_ms>
+```
+
+##### Examples:
+The below command subscribes to all the metrics starting with (prefix match) `org.apache.kafka.producer.record`
+with an interval of 10 seconds.
+
+```shell
+./bin/kafka-client-metrics.sh --bootstrap-server localhost:9092 --alter --name producer-record-metrics --metrics "org.apache.kafka.producer.record" --interval 10000
+```
+
+The below command subscribes to all the metrics with an interval of 10 seconds.
+
+```shell
+./bin/kafka-client-metrics.sh --bootstrap-server localhost:9092 --alter --name all-metrics --metrics "*" --interval 10000
+```
+
+### Create a Kafka Client
+
+Create a Kafka client that sends metrics to the broker. The client should be using the Kafka client
+version 3.7.0 or higher.
+
+#### Start console producer
+
+```shell
+./bin/kafka-console-producer.sh --broker-list localhost:9092 --topic test
+```
+
+### Access Client Metrics
+
+Produce messages to the topic `test` and observe the metrics in the Prometheus instance. Prometheus
+can be accessed at `http://localhost:9090`.
+
```

Comment:
I think it would be worth saying that the metrics will start with `kip_714_` in Prometheus so that people can start typing into the search expression box and have auto-complete show them the metric names.

Reply:
Done.